engine/dlib/src/dlib/job_thread.cpp

```diff
@@ -89,16 +89,19 @@ static void PutDone(JobThreadContext* ctx, JobItem* item)
 static void JobThread(void* _ctx)
 {
     JobThreadContext* ctx = (JobThreadContext*)_ctx;
-    while (dmAtomicGet32(&ctx->m_Run) != 0)
+    while (true)
     {
-        JobItem item;
+        JobItem item = {};
         {
             DM_MUTEX_SCOPED_LOCK(ctx->m_Mutex);
+
+            if (!ctx->m_Run)
+                break;
+
             while(ctx->m_Work.Empty())
             {
                 dmConditionVariable::Wait(ctx->m_WakeupCond, ctx->m_Mutex);
```

Comment:
Once we got here, the signal had already been sent which meant that the wait would then stall forever, as there wouldn't be any new signal coming.